app/src/main/java/com/example/gabriel/sociala/HomeActivity.java

```diff
@@ -189,4 +223,60 @@ public GridHolder(@NonNull View itemView) {
             textView = itemView.findViewById(R.id.tvCaption);
         }
     }
+
+
+
+
+
+    private void postPhoto() {
+
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M){
+            if (checkSelfPermission(Manifest.permission.READ_EXTERNAL_STORAGE)
+                    == PackageManager.PERMISSION_DENIED){
+                //permission not granted, request it.
+                String[] permissions = {Manifest.permission.READ_EXTERNAL_STORAGE};
+                //show popup for runtime permission
+                requestPermissions(permissions, PERMISSION_CODE);
+            }
+            else {
+                //permission already granted
+                pickImageFromGallery();
+            }
+        }
+        else {
+            //system os is less then marshmallow
+            pickImageFromGallery();
+        }
+
+    }
+
+
+    private void pickImageFromGallery() {
+        //intent to pick image
+        Intent intent = new Intent(Intent.ACTION_PICK);
+        intent.setType("image/*");
+        startActivityForResult(intent, IMAGE_PICK_CODE);
```

Comment:
Would selecting more then 1 img be incredibly hard here?

Reply:
> Would selecting more then 1 img be incredibly hard here?

I don't think so it's just because I don't have more time tonight. I have another due tomorrow. I'll work on multi photos tomorrow